packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -621,10 +615,15 @@ export function createSdkContext<
     __namespaceToApiVersionParameter: new Map(),
     __tspTypeToApiVersions: new Map(),
     __namespaceToApiVersionClientDefaultValue: new Map(),
+    examplesDirectory: context.options["examples-directory"],
     decoratorsAllowList: [...defaultDecoratorsAllowList, ...(options?.additionalDecorators ?? [])],
     previewStringRegex: options?.versioning?.previewStringRegex || /-preview$/,
   };
   sdkContext.sdkPackage = getSdkPackage(sdkContext);
+  for (const client of sdkContext.sdkPackage.clients) {
+    diagnostics.pipe(await handleClientExamples(sdkContext, client));
```

Comment:
Will any exception in handleClientExamples break the createSdkContext?

Does that mean if we don't care about example we don't need to wait this promise?

Reply:
all exception will be in diagnostics, will not block `createSdkContext`

the example will be added in each operation, so it is impossible to do sync way.